### ISSUE TITLE
Qwen3-8B and other models generate garbage output / repeat tokens (GGGGGG...) in llama.cpp via LM Studio Vulkan backend

### ISSUE THREAD

**Issue by fav-devs** [ggml-org/llama.cpp#13310]
### Name and Version

When using llama.cpp (via LM Studio with Vulkan backend and GGUF models), some models — specifically Qwen3-8B, Mistral, Hermes, and LLaMA — repeatedly output GGGGGGGGGGGG or loop indefinitely, regardless of prompt. This does not happen with Gemma models.

Model Format:
GGUF Q4_K_M, Q5_K_M, Q6_K
Tokenizer: auto-loaded by LM Studio

What I've Tried:
Adding <|im_start|> formatting for Qwen
Using [INST]...[/INST] for Mistral/LLaMA
Setting stop sequences
Lowering temp, top-k, top-p, repeat penalty
Testing via both UI and raw curl
Re-downloading models from official sources

Hypothesis:
Could be:
A tokenizer mismatch
Broken prompt template injection
Quantization compatibility issue with Vulkan
Missing EOS/stop condition on some models

### Operating systems

Windows

### GGML backends

Vulkan

### Hardware

LM Studio v0.3.15 (Build 11)
llama.cpp backend via Vulkan
GPU: AMD Radeon RX 5700 (7.98 GB VRAM)
RAM: 32 GB
OS: Windows 10 x64

### Models

Qwen3-8B.Q4_K_M.gguf https://huggingface.co/lmstudio-community/Qwen3-8B-GGUF
Qwen2.5-7B-Instruct-1M-GGUF  https://huggingface.co/lmstudio-community/Qwen2.5-7B-Instruct-1M-GGUF

### Problem description & steps to reproduce

Load Qwen3-8B.Q4_K_M.gguf or Mistral-7B-Instruct-v0.1.Q4_K_M.gguf

Run this prompt:
<|im_start|>user
Write a long message about the sun.<|im_end|>
<|im_start|>assistant

Output returns:
GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG...

✅ Works Fine:
gemma-3-4b-it-qat
gemma-3-12b-it-qat

### Relevant log output

```shell
write a long message

qwen3-8b@q3_k_l
Thoughts
Thought for 18.52 seconds

Okay, the user asked me to write a long message again. Let me check the history to understand what they need.

In the previous interaction, the user first said "hi" and then started typing a long string of "io". I responded with an HTML landing page example. Then they asked for a long message, which I provided as a reflective text on resilience, connection, and imperfection. Now they're asking again for a long message.

Wait, maybe the user is testing if I can generate longer content or perhaps they want another similar response. But since the last message was already quite lengthy, maybe they want something different this time? Or perhaps they made a mistake in their query. Alternatively, maybe they want more examples of long messages on various topics.

I need to ensure that my response is helpful and not redundant. The user might be looking for another example, but it's possible they just want the same type of message again. To avoid repetition, I should check if there's a specific context or theme they have in mind. However, since they haven't provided more details, I'll proceed by offering a new, different long message that expands on previous themes but introduces new elements.

Alternatively, maybe they want a message about a different topic like technology, nature, or personal growth. Let me think of another angle. Perhaps focusing on gratitude, mindfulness, or the journey of self-discovery. Ensuring that the message is well-structured, with clear sections and a coherent flow. Also, making sure it's engaging and thought-provoking.

I should also consider the user's possible needs. If they're working on a project, maybe they need content for a blog post or a speech. Or perhaps they want to explore different aspects of human experience. Including elements like personal anecdotes, philosophical reflections, and practical advice would make the message more relatable.

Another approach is to structure the message with clear sections, such as an introduction, body paragraphs on various themes, and a conclusion. Adding emotional depth and encouraging self-reflection can make the message impactful. I'll need to maintain a balance between being comprehensive and not overwhelming the reader.

Also, considering the previous response was about resilience and imperfection, maybe this time focusing on gratitude or embracing change. Ensuring that each paragraph transitions smoothly and maintains a consistent tone. Using rhetorical questions, metaphors, and personal insights can enhance the message's depth.

I should also check for any possible repetition and ensure that new ideas are introduced. Maybe include elements like mindfulness practices, overcoming fear, or finding purpose in life. Making sure the message is both informative and inspirational.

In summary, I'll create a new long message with a different theme, possibly focusing on gratitude and mindfulness, structured into sections with clear explanations and personal reflections to provide value to the user.

A Long Message: The Art of Being Present

Let me tell you something that has been on my mind for weeks now—something I’ve come to realize is both simple and profound. It’s about being present. Not in the way we often misunderstand it (as a checklist item, "I should meditate 10 minutes today"), but as a deep, unspoken truth: life is not about chasing what's next; it's about holding still enough to notice what is now.

You see, I’ve spent so much of my life in a hurry—to grow faster, achieve more, fix everything that feels broken, and avoid the uncomfortable truths of who I am. But here’s the catch: hurrying has always been a kind of denial. It’s an avoidance of the present moment, a refusal to sit with what is, instead of what we think it should be. And yet, every time I’ve forced myself to pause—whether for a breath, a cup of tea, or simply watching the4 leaves fall from a tree—I’ve discovered something remarkable: the world is alive in ways I never imagined.

Take this morning, for example. The sun was just rising when I stepped outside. The air felt fresh, crisp with dew, and the sky had thatotros hues of lavender and gold that made me stop and breathe. A sparrow landed on the railing nearby, chirping a melody so pure it seemed like it had been waiting for me to listen. And in that moment, I was not a busy person rushing to work, nor an anxious mind worrying about deadlines or emails. I was simply here.

This is what presence looks like—it’s not about grand gestures or elaborate rituals. It’s the quiet act of noticing, the willingness to let go of control, and the courage to embrace uncertainty. It’s when you realize that even in chaos, there is rhythm; even in silence, there is sound; and even in loneliness, there is connection.

Let me share a story from my own life. A few years ago, I was going through one of those periods where everything felt off-kilter. My work was demanding, my relationships were strained, and I couldn’t shake the feeling that I was failing at being "enough." One evening, I decided to walk alone in the woods near my home. It wasn’t a planned trip—it just happened. As I wandered through the forest, I found myself drawn to a small stream. The water was clear, cool, and it flowed with an effortless grace. I sat down on the rocks and watched it for what felt like hours. No phone, no distractions, no need to "do" anything. Just being.

And in that stillness, something shifted. I didn’t find answers. I didn’t solve problems. But I found peace. I realized that my fear of not having enough wasn’t about the future—it was about the present. The stream taught me that even when things feel turbulent, there is a natural order to it. It’s not about controlling the flow; it’s about learning how to move with it.

This brings me to something I’ve been thinking about lately: the power of small, consistent actions. We often underestimate what happens when we show up daily, even in tiny ways. A morning walk, a few minutes of gratitude journaling, or simply allowing yourself to feel emotions without judgment—all these acts are not just habits; they’re rituals of presence. They remind us that life is not about perfection, but about showing up again and again.

And yet, I know this isn’t easy. The world around us is full of noise—social media, deadlines, comparisons, and the constant pressure to "keep going." It can be overwhelming. But here’s what I’ve learned: presence is a choice. It doesn’t require grand gestures or extraordinary circumstances. It just requires intention. It’s about choosing to notice the beauty in small things, like the way sunlight filters through leaves, the warmth of a hand on your shoulder, or the quiet comfort of being with someone who truly sees you.

Let me also talk about connection—because presence is not just about ourselves; it’s about others. We are inherently social beings, wired to connect. Yet so often, we get lost in our own heads, forgetting that we are part of something larger. I’ve seen how meaningful it can be to simply sit with someone, listen without judgment, or share a meal without the distractions of screens. These moments aren’t just about conversation; they’re about being with each other, in all our imperfections.

And here’s the truth: imperfection is not something to fix. It’s something to embrace. I used to think that life was about becoming better—about improving, growing, and achieving. But now I see it differently. Life is about being—about showing up as you are, even when you’re messy, even when you don’t have all the answers. It’s about allowing yourself to stumble, to fall, and to rise again without shame. Because every time we do that, we become more resilient, more compassionate, and more whole.

This is not to say there aren’t challenges—because life is hard, and it doesn’t always make sense. But I’ve come to believe that the most meaningful moments are often the ones we don’t plan. The unexpected call from a friend, the sudden realization during a walk, or the quiet joy of watching a child laugh. These are the things that remind us why we’re here—to love, to connect, and to findGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG

29.99 tok/sec

•

1981 tokens

•

7.24s to first token

•

Stop reason: User Stopped
```

**Comment by lcarrere** [comment]
I’m also observing this, with all produced logits having NaN value, with different Nvidia GPUs

**Comment by jeffbolznv** [comment]
Can somebody share a command line to reproduce this with llama-cli? I tried yesterday but wasn't able to reproduce it.

**Comment by characharm** [comment]
I have a similar issue with llama-server. The launch flags are --jinja -c 15000 -ngl 99 --port 8000, nothing special in the parameters, so at first I thought the problem was with my setup. I'm using two GPUs – Intel and AMD – and I've noticed that after a short period of inactivity, the model gets unloaded from the Intel GPU into main RAM, and then reloaded back when a new request comes in. Often, the issue with the 'GGGGGGGGGGG' output shows up after a short idle period, usually after two or three good responses. But sometimes it happens right away. Restarting the server usually fixes it.

**Comment by starble-dev** [comment]
I have the same issue too, however this is for GLM-4-32B using Q_4_K_S from [bartowski's quant](https://huggingface.co/bartowski/THUDM_GLM-4-32B-0414-GGUF), however some investigation at different HF discussions appears the issue specifically for GLM-4-32B comes from batching as using the settings "-b 8 -ub 8" for llama-server appears to fix my issue. This happens to me both for Vulkan and ROCm. Not sure if this is actually a genuine fix but I found this out via [this discussion](https://huggingface.co/unsloth/GLM-4-32B-0414-GGUF/discussions/5) in HF.

Changing the batch size does fix the issue in both Vulkan and ROCm. Weirdly enough this doesn't seem to happen if quanting the KV-cache so it's probably a float issue.

**Comment by stduhpf** [comment]
>  however some investigation at different HF discussions appears the issue specifically for GLM-4-32B comes from batching as using the settings "-b 8 -ub 8" for llama-server appears to fix my issue. 

Thanks for pointing this out, though it's not really a fix, but a workaround that comes at a big cost in prompt processing speed.

Probably related to this issue, I get a lot of failures with test-backend-ops, with f16 ADD,SUB,MUL, and DIV ops. I'll try to pin-point which commit introduced these failing tests to see if reverting it fixes this issue.

**Comment by fav-devs** [comment]
Hi there, everyone, I discovered a weird solution after looking at what could make gemma special from all the other models i tried i noticed the gemma's quat was Q4_0 and all the other models were all Q4_K_M so i downloaded a Q3_K_L version of the Qwen model and so far i have not had any issues

**Comment by characharm** [comment]
It turns out my problem is most likely not related to this bug. I made a small SYCL utility that constantly pings a small area of memory on the Intel card, preventing VRAM from being swapped to RAM. The server has been running for more than 24 hours without errors.

**Comment by 0cc4m** [comment]
Thank you for bisecting it. @netrunnereve maybe we missed something in the GCN tune.

**Comment by netrunnereve** [comment]
@nasrally I went and tried a couple long generations with Qwen3 4B (that's the only Qwen that I have with me right now) and couldn't get it to repeat like that. Are you seeing this with other models like Mistral and Llama as well?

I'm going to download bartowski/Qwen_Qwen3-14B-GGUF:Q6_K to try out but in the meantime please run a `test-backend-ops` and  do some tests with FP16 turned off using GGML_VK_DISABLE_F16=1. I mean I doubt that this is a FP16 issue but it's worth a try.

**Comment by 0cc4m** [comment]
@nasrally Which mesa version are you using?

With a Radeon Pro VII on Mesa 24.2.8 I have working Qwen3-14B, but Qwen3-30B-A3B is not working with FP16 enabled. If I disable FP16, it starts working.

With Mesa 25.1.0 Qwen3-14B is completely broken, regardless of fp16. Qwen3-30B-A3B is also broken either way.

I'm not sure if this is the same problem you have, but it's making testing difficult for me. This isn't the first time I'm having trouble with it, mesa has had a streak of breaking my Pro VII and I'm not sure what is causing this. On amdvlk I don't see any of these problems.

**Comment by 0cc4m** [comment]
Look at the device info like, if it mentions `(RADV ...` in the device name, it's radv. If not, it's amdvlk or vulkan-amdgpu-pro.

**Comment by 0cc4m** [comment]
amdvlk no longer supports GCN, so you would need an old version.

**Comment by netrunnereve** [comment]
Those `test-backend-ops` results are terrible, especially with FP16 on. I would expect a lot of models to have wrong results on your system. Now I went through the code again and I don't see anything in my 5175 which would mess up FP16 but not FP32, and the only thing I can think of if that maybe the use of 256 rather than 128 threads for a 64x64 block causes some precision differences when we add up the FP16 numbers.

I also went and tested out the Q6_K 14B Qwen and couldn't get it to repeat like that or generate garbage, though it seems like that model doesn't set the EOS properly as it doesn't stop after finishing it's response. So it'll think and answer the question all over again, but it doesn't mess up the output with gibberish. Interesting enough the prompt processing speed is super slow on my W8100 + RX 470 at 15 t/s, but that's another issue and I'm not going to worry about that here.

BTW if you aren't sure if an issue is due to the model or the backend you can try running it on CPU only with the same prompt. There's no guarantees of course but the CPU implementation should be correct... right?

**Comment by 0cc4m** [comment]
No, this is a RADV problem, related to fp16 on GCN. Someone has to open an issue upstream.

**Comment by netrunnereve** [comment]
The strange part though is that the results are apparently more coherent before 5175, even though test-backend-ops has the same number of errors. At the core this is a driver issue but the tune might have some effect as well.

**Comment by bjodah** [comment]
> I see garbage with all Qwen3 models I ran which are:
> 
>     * bartowski/Qwen_Qwen3-14B-GGUF:Q4_K_L
> 
>     * bartowski/Qwen_Qwen3-14B-GGUF:Q6_K
> 
>     * unsloth/Qwen3-30B-A3B-GGUF:Q4_K_XL
> 
>     * mradermacher/Qwen3-30B-A4.5B-12-Cooks-GGUF:Q4_K_S
> 
> 
> All of them output garbage. Latest version I for sure know works is 5141 (haven't tried much else though)

I see the `GGGGGGGG...` issue on `--hf-repo bartowski/Qwen_Qwen3-14B-GGUF:Q6_K_L` too (I'm using CUDA backend). If I switch to bartowski's Q8_0 the problem seems to go away. For my particular test case, unsloth's Q6_K_XL also works.

**Comment by ghost** [comment]
After a bit of investigation, I was able to narrow it down to -np > 1 on the following setup:

Llama version: ghcr.io/ggml-org/llama.cpp  server-cuda     8489cd1c2ecd, this should be b5452
Model: Qwen3 32bit Q4_K_M from Ollama repository (not the newest one, but the one before)
GPUs: 2x V100 16 GB

Tried with both 16 and 8k context, with np=1, 2, and 4. The np=1 works fine for both contexts, none of the other options work.

**Comment by longtian** [comment]
same problem here, I was able to solve it by decreasing `-ub`

```
-ub,   --ubatch-size N                  physical maximum batch size (default: 512)
```

to 128 on Jetson Orin Nano Developer Super (firmware 36.4.4)
